### DISCUSSION TITLE
NavigateAsync has no useModalNavigation and animated parameters - Prism.DryIoc.Maui

### DISCUSSION THREAD

**Discussion by HelenMamalaki** [PrismLibrary/Prism#2928] [.NET MAUI] [answered]
Hi, I have come across this closed issue #2225 about these parameters no longer being optional which was Closed in 2020 - but I don't see an overload method that takes all 4 parameters in the latest Prism.DryIoc.Maui release - from Aug 2022- (I know that sponsors get access to newer version but I assume the free tier will receive an update at some point?) The only issue I'm having just now is these missing parameters.
Am I missing a reference to something? Any suggestions would be greatly appreciated. :)
<PackageReference Include="Prism.DryIoc.Maui" Version="8.1.273-pre" />

**Comment by dansiegel** [comment, marked as answer]
Yes this was removed because it should be passed in the NavigationParameters.

https://github.com/PrismLibrary/Prism/blob/bf0966af97a7dd16141ed3b3c3cfc784ff5e0cb7/src/Maui/Prism.Maui/Navigation/KnownNavigationParameters.cs#L3-L29

**Reply by HelenMamalaki** [reply]
Thank you for the quick reply! I have tried to switch off animation and tried modal navigation true/false but nothing seems to change. All I have to do is pass them as parameters? Do I need to do something in the receiving method?
![image](https://github.com/PrismLibrary/Prism/assets/4314194/7f63ec08-d6a1-4c55-9453-684ccc7fa2d4)
I have a Windows Desktop app without a back button  (net7-windows 10.0.19041) and the page animation makes everything look like a modal page.

**Reply by jameslavery** [reply]
I'm seeing the same or a similar problem with MAUI on Windows.

I'm trying to turn animation off with the following code:

```
        var navParams = new NavigationParameters()
        {
            {KnownNavigationParameters.Animated,false},
            {KnownNavigationParameters.UseModalNavigation, false },
        };

        _navigationService.NavigateAsync(nameof(SecondPage),navParams);
```
but the navigation still animates (the page appears to slide to the left when navigating). No matter what I do, the animation remains.

In a MAUI application without Prism, this navigation call (the _false_ being the animation argument):

```Navigation.PushModalAsync(new` SecondPage(),false);```

results in navigation with no animation.

Is this a Prism bug/feature, or am I conflating MAUI navigation and Prism navigation?

**Reply by dansiegel** [reply]
If you are seeing an animation with it explicitly set to false it's definitely a bug. If you can submit a sample app that reproduces the issue and open an issue we can look into it.

**Reply by jameslavery** [reply]
Thanks - #3031 raised, with reproduction (Windows only) attached.
